addon/components/sl-split-grid.js

```diff
@@ -0,0 +1,564 @@
+     *
+     * @function getSortedColumn
+     * @returns  {object} - The definition for the currently sorted column
+     */
+    getSortedColumn: function() {
+        var columns           = this.get( 'columns' ),
+            sortedColumnTitle = this.get( 'sortedColumnTitle' );
+
+        if ( sortedColumnTitle ) {
+            for ( var i = 0; i < columns.length; i++ ) {
+                if ( Ember.get( columns[ i ], 'title' ) === sortedColumnTitle ) {
+                    return columns[ i ];
+                }
+            }
+        }
+    },
+
+    /**
+     * Callback to the list content scrolling, which is responsible for
+     * determining when triggering nextPage is necessary by checking the
+     * scroll location of the content
+     *
+     * @function handleListContentScroll
+     * @param    {jQuery.Event} event - The scroll trigger event
+     * @returns  {void}
+     */
+    handleListContentScroll: function( event ) {
+        var listContent  = this.$( event.target ),
+            scrollBottom = listContent.scrollTop() + listContent.height();
+
+        if ( scrollBottom >= this.get( 'nextPageScrollPoint' ) && !this.get( 'pendingData' ) ) {
+            this.requestNextPage();
+        }
+    },
+
+    /**
+     * Whether the content has more available data to page in
+     *
+     * @function hasMorePages
+     * @returns  {boolean} - True if more content pages are available
+     */
+    hasMorePages: function() {
+        return this.get( 'content.length' ) < this.get( 'totalCount' );
+    }.property( 'content.length', 'totalCount' ),
+
+    /**
+     * Trigger the bound `nextPage` action for more data
+     *
+     * @function requestNextPage
+     * @returns  {void}
+     */
+    requestNextPage: function() {
+        if ( this.get( 'hasMorePages' ) ) {
+            this.setProperties({
+                'pendingData'         : true,
+                'nextPageScrollPoint' : this.$( '.list-pane .content' )[ 0 ].scrollHeight
+            });
+
+            this.sendAction( 'nextPage' );
+        }
+    },
+
+    /**
+     * Trigger calculation of content height for the list pane and detail pane
+     *
+     * @function updateContentHeight
+     * @returns  {void}
+     */
+    updateContentHeight: function() {
+        this.updateDetailContentHeight();
+        this.updateListContentHeight();
+    },
+
+    /**
+     * Calculate and set the content height for the detail pane
+     *
+     * @function updateDetailContentHeight
+     * @returns  {void}
+     */
+    updateDetailContentHeight: function() {
+        var viewportHeight     = Ember.$( window ).innerHeight(),
+            topPosition        = this.$().position().top,
+            gridHeaderHeight   = parseInt( this.$( '.grid-header' ).css( 'height' ) ),
+            detailHeaderHeight = parseInt( this.$( '.detail-pane header' ).css( 'height' ) ),
+            detailFooterHeight = parseInt( this.$( '.detail-pane footer' ).css( 'height' ) ) || 0,
```

Comment:
Can this make use of http://api.jquery.com/height?  Is it more performant to do so?